pom.xml

```diff
@@ -64,6 +64,7 @@
     <checkstyle.violation.ignore>RedundantThrows,NewlineAtEndOfFile,ParameterNumber,MethodLength,FileLength,ModifierOrder</checkstyle.violation.ignore>
 
     <mavenVersion>3.0</mavenVersion>
+    <javaVersion>7</javaVersion>
```

Comment:
This really be a separate issue.

Reply:
Then we should do that first, and it would block this issue. We use the `java.nio.file` in this pull request so we can't continue before that is resolved.

Reply:
Correct, please do so!

Reply:
See [MSHARED-826](https://issues.apache.org/jira/browse/MSHARED-826). I'll pick that one up soon.

Reply:
That one is now merged, and I removed the commits in this PR after having rebased the branch on master.